configure.ac

```diff
@@ -236,7 +236,14 @@ AC_CHECK_PROG(YASM_CHECK, yasm, yes)
 if test x"$YASM_CHECK" = x"yes"; then
   if yasm -f elf64 src/common/crc32c_intel_fast_asm.S -o /dev/null; then
      echo 'we have a modern and working yasm'
-     if test `arch` = "x86_64" ; then
+     # FreeBSD has uname -p not arch(1), so do not exec arch
+     arch_x86_64=0
+     if test `uname -p` = "amd64"; then
+	arch_x86_64=1
+     elif test  -x /usr/bin/arch -a `arch` = "x86_64" ; then
```

Comment:
On 22-3-2016 04:27, Kefu Chai wrote:
> In configure.ac
> <https://github.com/ceph/ceph/pull/7515#discussion_r56932924>:
> 
>> @@ -236,7 +236,14 @@ AC_CHECK_PROG(YASM_CHECK, yasm, yes)
>>  if test x"$YASM_CHECK" = x"yes"; then
>>    if yasm -f elf64 src/common/crc32c_intel_fast_asm.S -o /dev/null; then
>>       echo 'we have a modern and working yasm'
>> -     if test `arch` = "x86_64" ; then
>> +     # FreeBSD has uname -p not arch(1), so do not exec arch
>> +     arch_x86_64=0
>> +     if test `uname -p` = "amd64"; then
>> +	arch_x86_64=1
>> +     elif test  -x /usr/bin/arch -a `arch` = "x86_64" ; then
> 
> why you'd like to check a specific path for |arch| and call it just
> using "arch" (i.e. without specifying the full path?

The test is there to make sure that arch is there and executable.
Since there are platforms that do not have arch, and then selection for
YASM really haywire.

Perhaps it even call for another else that will print an warning that
YASM type selection was not completed correctly.
That could fail now atm.